allennlp/data/iterators/data_iterator.py

```diff
@@ -225,19 +225,35 @@ def _memory_sized_lists(self,
             yield list(iterator)
 
 
-    def _ensure_batch_is_sufficiently_small(self, batch_instances: Iterable[Instance]) -> List[List[Instance]]:
+    def _ensure_batch_is_sufficiently_small(
+            self,
+            batch_instances: Iterable[Instance],
+            excess: deque) -> List[List[Instance]]:
         """
-        If self._maximum_samples_per_batch is specified, then split the batch into smaller
-        sub-batches if it exceeds the maximum size.
+        If self._maximum_samples_per_batch is specified, then split the batch
+        into smaller sub-batches if it exceeds the maximum size.
+
+        Any excess passed in will be used first. When the method returns excess
```

Comment:
Could you mention explicitly in this docstring that `excess` is mutated in-place.

Reply:
Cleaned up the language and left a big warning.